meta-balena-common/recipes-containers/resin-supervisor/resin-supervisor/resin-data.mount

```diff
@@ -2,7 +2,6 @@
 Description=resin-data bind mount
 Requires=resin-data.service
 After=resin-data.service
-DefaultDependencies=no
```

Comment:
Is there a specific default dependency we are hoping to inherit here that we didn't want/need previously?

https://www.freedesktop.org/software/systemd/man/systemd.mount.html#Default%20Dependencies

Reply:
It's not obvious to me either when reading the documentation. However, if you get the current master and you remove the patch that adds the `DefaultDependency` and you reboot several times you will eventually encounter a dependency failure on boot:
```
[DEPEND] Dependency failed for resin-data bind mount.
```
This does not happen without the default dependencies. I suspect it's the dependencies on local file system targets but I'm not sure.
The fact that removing default dependencies made it work reliably masked the real problem with the resin-data mounts which we have now found out.
I went back, removed this, performed a 50 loops reboot cycle and did not hit the dependency failed problem. As I can see no reason to leave this I decided to remove it, but we should keep an eye on this when testing over multiple platforms.